src/_pytest/python.py

```diff
@@ -1552,7 +1595,21 @@ def get_best_relpath(func) -> str:
         loc = getlocation(func, invocation_dir)
         return bestrelpath(invocation_dir, Path(loc))
 
-    def write_fixture(fixture_def: fixtures.FixtureDef[object]) -> None:
+    def write_item(item: nodes.Item) -> None:
+        fixturedefs = _get_fixtures_per_test(item)
+        if not fixturedefs:
+            # This test item does not use any fixtures.
+            # Do not write anything.
+            return
+
+        tw.line()
+        tw.sep("-", f"fixtures used by {item.name}")
+        tw.sep("-", f"({get_best_relpath(item.function)})")  # type: ignore[attr-defined]
+
+        for fixturedef in fixturedefs:
+            write_fixture(fixturedef)
+
+    def write_fixture(fixture_def: FixtureDef[object]) -> None:
         argname = fixture_def.argname
         if verbose <= 0 and argname.startswith("_"):
             return
```

Comment:
This condition needs to be moved into `write_item` _before_  the occurrence of the terminal writes

```python
     tw.sep("-", f"fixtures used by {item.name}")
     tw.sep("-", f"({get_best_relpath(item.function)})") 
```
and adapted to filter `fixturedefs` of such private fixtures (assuming verbosity less than or equal to zero).

Otherwise there is a circumstance where all an item's fixtures have been excluded from the output yet the terminal still produces output for that item.

For example, if all of `test_private_fixtures`'s fixtures start with `_` and verbosity is zero or less, then no fixtures will be shown but `fixtures used by test_private_fixtures` etc will be shown.

TODO: 
- add tests for fixtures starting with `_`
- adapt condition to conditionally filter the `fixturedefs` list

Reply:
I have confirmed that in this PR private fixtures are not shown when verbosity is zero or less but the _fixtures used by . . ._ is still printed. Is this desired behaviour?
I don't know. It _is_ how _pytest_ currently handles reporting on private fixtures.

```python
import pytest

# DEFINE FIXTURES
#################

@pytest.fixture
def _private():
    """Private fixture"""
    pass

@pytest.fixture
def public():
    """Public fixture."""
    pass

# TESTS
#################

def test_private_fixture(_private):
    pass

def test_public_fixture(public):
    pass

@pytest.mark.parametrize("pseudo", [1])
def test_pseudo_fixture(pseudo):
    pass

def test_no_fixture():
    pass
```

Running `pytest --fixtures-per-test` on the current version of pytest shows:
```
----------- fixtures used by test_private_fixture ------------
-------------------- (test_private.py:14) --------------------

------------ fixtures used by test_public_fixture ------------
-------------------- (test_private.py:17) --------------------
public -- test_private.py:9
    Public fixture.

---------- fixtures used by test_pseudo_fixture[1] -----------
-------------------- (test_private.py:20) --------------------
pseudo -- .../_pytest/python.py:1112
    no docstring available
```

I'm proposing it shows:
```
--------------------- fixtures used by test_public_fixture ----------------------
-------------------------- (test_fixture_marker.py:23) --------------------------
public -- test_fixture_marker.py:12
    Public fixture.
```

Notable differences:
- pseudo fixtures in tests are ignored
- a test with only private fixtures is treated like a test with no fixtures at verbosity zero or less (i.e. the test is not reported on at all)

The output this PR currently produces is:
```
--------------------- fixtures used by test_private_fixture ---------------------
-------------------------- (test_fixture_marker.py:20) --------------------------

--------------------- fixtures used by test_public_fixture ----------------------
-------------------------- (test_fixture_marker.py:23) --------------------------
public -- test_fixture_marker.py:12
    Public fixture.
```